src/display/editor/highlight.js

```diff
@@ -154,29 +161,43 @@ class HighlightEditor extends AnnotationEditor {
       }
     }
 
-    const innerMargin = 1.5;
-    this.#focusOutlines = highlight.getFocusOutline(
+    const extraThickness = 1.5;
+    this.#focusOutlines = highlight.getNewOutline(
       /* Slightly bigger than the highlight in order to have a little
          space between the highlight and the outline. */
-      HighlightEditor._defaultThickness + innerMargin
-    );
-    this.#outlineId = this.parent.drawLayer.highlightOutline(
-      this.#focusOutlines
+      this.#thickness / 2 + extraThickness,
+      /* innerMargin = */ 0.0025
     );
+
+    if (highlightId >= 0) {
+      this.#outlineId = this.parent.drawLayer.highlightOutline(
+        this.#focusOutlines
+      );
+    } else {
+      this.parent?.drawLayer.updateLine(this.#outlineId, this.#focusOutlines);
+    }
     this.#lastPoint = lastPoint;
   }
 
+  /** @inheritdoc */
   static initialize(l10n, uiManager) {
     AnnotationEditor.initialize(l10n, uiManager);
     HighlightEditor._defaultColor ||=
       uiManager.highlightColors?.values().next().value || "#fff066";
+    document
+      .getElementById("editorHighlightThickness")
+      .style.setProperty("--pdf-to-css-units", PixelsPerInch.PDF_TO_CSS_UNITS);
```

Comment:
Sorry, but this feels quite "hacky". What are we actually trying to do here, since it's not clear to me why the toolbar UI-elements must change with this value?

---

Note that we try to avoid this sort of inline DOM-element lookup e.g. throughout the viewer code. Also, this would likely break if used *outside* of the default viewer.
```suggestion
    document
      .getElementById("editorHighlightThickness")
      ?.style.setProperty("--pdf-to-css-units", PixelsPerInch.PDF_TO_CSS_UNITS);
```

Reply:
My idea was to have the same dimensions as the ones we've at 100%:
![image](https://github.com/mozilla/pdf.js/assets/5641725/00efaa15-1427-4c46-b9a6-c06f59ed101a)
maybe it's useless... wdyt ?

Reply:
But that means that it'll only be "correct" at a single zoom-level, right?
To me that does seem a little inconsistent, and unless we always adapt those sizes to the current zoom-level I do wonder if this is necessary?

Perhaps we can skip doing this for now, and check with UX if such a feature is wanted?